src/witness/oracle.rs

```diff
@@ -552,8 +552,35 @@ pub fn create_artifacts_from_tracer<
         // wherever we have this marker we should look at the tail of the item right before it
         let pos = log_position_mapping[&rollback_tail_marker];
         let tail = if pos == -1 {
-            // empty
-            global_end_of_storage_log
+            if frame_index > 1 {
```

Comment:
FYI - this is more than 'adding tests' - please update the PR description

Reply:
@mm-zk you are right. Even more, proposed changes do not fix corresponding issue. I will rewrite this part of logic